### PR TITLE
Updating Wazuh DB command to update vulnerabilities status by their type

### PR DESCRIPTION
**Pull request by pereyra-m**
|Related issue|
|---|
|#7989|

## Description

This PR creates a new Wazuh-DB helper method to update the vulnerabilities in `vuln_cves` table according their type.
Also, some UT were updated considering the changes in the **wdb_agents_update_status_vuln_cves()** method.

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
- [x] Source installation

- Memory tests for Linux
  - [x] Scan-build report

### CODE REVIEW

**Comment by pereyra-m** [comment]
### Tests

The command was tested using sample data and sending requests to the Wazuh-DB socket.

- The initial state

cve1

- The update command

cve2

- The result

cve3

- Another command with its result

cve4


### Scan build

There weren't any new bugs reported.

scan_build_no_bugs

File: src/wazuh_db/wdb_parser.c [hunk 1]
```diff
@@ -6149,14 +6149,20 @@ int wdb_parse_agents_vuln_cves_update_status(wdb_t* wdb, char* input, char* outp
...
-        if (!cJSON_IsString(old_status) || !cJSON_IsString(new_status)) {
-            mdebug1("Invalid vuln_cves JSON data when updating status value. Not compliant with constraints defined in the database.");
-            snprintf(output, OS_MAXSTR + 1, "err Invalid JSON data, missing required fields");
-        }
+        cJSON* type = cJSON_GetObjectItem(data, "type");
 
-        else {
-            ret = wdb_agents_update_status_vuln_cves(wdb, old_status->valuestring, new_status->valuestring);
+        if (cJSON_IsString(type) && cJSON_IsString(new_status) && !cJSON_IsString(old_status)) {
+            ret = wdb_agents_update_status_vuln_cves(wdb, NULL, new_status->valuestring, type->valuestring);
```
Comment (palaciosjeremias) [inline src/wazuh_db/wdb_parser.c:6155]: We can simplify this by obtaining NULL or valuestring for each parameter before, and making only one call with only one error handling:
```
const char new_status_str = cJSON_IsString(new_status) ? new_status->valuestring : NULL;
const char old_status_str = cJSON_IsString(old_status) ? old_status->valuestring : NULL;
const char type_str = cJSON_IsString(type) ? type->valuestring : NULL;
wdb_agents_update_status_vuln_cves(wdb, old_status_str, new_status_str, type_str);
```

or even simpler: Probably valuestring is **NULL** when **!cJSON_IsString()** and we can use the value directly

**Review by palaciosjeremias — APPROVED** [review]

**Review by palaciosjeremias — APPROVED** [review]
LGTM

File: src/wazuh_db/wdb_agents.c [hunk 2]
```diff
@@ -94,21 +94,37 @@ cJSON* wdb_agents_insert_vuln_cves(wdb_t *wdb,
     return result;
 }
 
-int wdb_agents_update_status_vuln_cves(wdb_t *wdb, const char* old_status, const char* new_status) {
+int wdb_agents_update_status_vuln_cves(wdb_t *wdb, const char* old_status, const char* new_status, const char* type) {
+    sqlite3_stmt* stmt = NULL;
 
-    bool update_all = (strcmp(old_status, "*") == 0);
+    if (old_status && new_status && !type) {
```
Comment (DProvinciani) [inline src/wazuh_db/wdb_agents.c:100]: I suggest removing the last part of the condition `&& !type`. In this block of code within the `if` statement you will use only `old_status` and `new_status`. The `type` will not be used at all so it wouldn't be verified.

File: src/wazuh_db/wdb_agents.c [hunk 3]
```diff
@@ -94,21 +94,37 @@ cJSON* wdb_agents_insert_vuln_cves(wdb_t *wdb,
...
+        }
+
+        if (update_all) {
+            sqlite3_bind_text(stmt, 1, new_status, -1, NULL);
+        } else {
+            sqlite3_bind_text(stmt, 1, new_status, -1, NULL);
+            sqlite3_bind_text(stmt, 2, old_status, -1, NULL);
+        }
     }
+    else if (!old_status && new_status && type){
```
Comment (DProvinciani) [inline src/wazuh_db/wdb_agents.c:116]: Same here for the `old_status`.

**Review by DProvinciani — APPROVED** [review]
Looks good! Just some minor suggestions. The decision to change it or not is up to you.

File: src/wazuh_db/wdb_parser.c (hunk 1 above)
Comment (pereyra-m) [reply src/wazuh_db/wdb_parser.c]: Thank you. The method was updated

File: src/wazuh_db/wdb_agents.c (hunk 2 above)
Comment (pereyra-m) [reply src/wazuh_db/wdb_agents.c:100]: I'd rather leave the extra check, just to be sure the query isn't called with three parameters and generate an unexpected behavior.

File: src/wazuh_db/wdb_agents.c (hunk 3 above)
Comment (pereyra-m) [reply src/wazuh_db/wdb_agents.c:116]: I'd leave it here too.

File: src/wazuh_db/wdb_agents.c (hunk 2 above)
Comment (DProvinciani) [reply src/wazuh_db/wdb_agents.c:100]: Good! So let's keep it.

**Pull request merged**